sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```
@@ -209,28 +208,36 @@ private static PlacementRule parseRow(StringMatcher taskFilter, List<String> row
-        public PlacementRule run(StringMatcher taskFilter, String fieldName, String operatorName,
-                                 Optional<String> parameter) throws IOException;
+        public PlacementRule run(
+                StringMatcher taskFilter,
+                String fieldName,
+                String operatorName,
+                Optional<String> parameter) throws IOException;
     }
 
+
     /**
      * {@code UNIQUE} tells Marathon to enforce uniqueness of the attribute across all
      * tasks of a given type. For example the following constraint ensures that there is only one app task running
      * on each host for some task type: {@code [["hostname", "UNIQUE"]]}
      */
     private static class UniqueOperator implements Operator {
-        public PlacementRule run(StringMatcher taskFilter, String fieldName, String operatorName,
-                                 Optional<String> ignoredParameter) {
-            if (isHostname(fieldName)) {
-                return new MaxPerHostnameRule(1, taskFilter);
-            } else {
-                // Ensure that:
-                // - Task sticks to nodes with matching fieldName defined at all (AttributeRule)
-                // - Task doesn't exceed one instance on those nodes (MaxPerAttributeRule)
-                StringMatcher matcher = RegexMatcher.createAttribute(fieldName, ".*");
-                return new AndRule(
-                        AttributeRuleFactory.getInstance()
-                                .require(matcher), new MaxPerAttributeRule(1, matcher, taskFilter));
+        public PlacementRule run(
+                StringMatcher taskFilter,
+                String fieldName,
+                String operatorName,
+                Optional<String> ignoredParameter) {
+
+            switch (PlacementUtils.getField(fieldName)) {
+                case HOSTNAME:
+                    return new MaxPerHostnameRule(1, taskFilter);
+                case ATTRIBUTE:
+                default:
```

should the default case throw an `UnsupportedOperationException`?